### PR TITLE
Copy butler credentials into user home space

### PR DESCRIPTION
**Pull request by athornton**

### CODE REVIEW

File: jupyterlab/runlab.sh [hunk 1]
```diff
@@ -16,6 +16,31 @@ function setup_git() {
     fi
 }
 
+function copy_butler_credentials() {
+    # Copy the credentials from the root-owned mounted secret to our homedir,
+    #  set the permissions accordingly, and repoint the environment variables.
```
Comment (SimonKrughoff) [inline jupyterlab/runlab.sh:21]: Single extra space at beginning of comment

File: jupyterlab/runlab.sh [hunk 2]
```diff
@@ -16,6 +16,31 @@ function setup_git() {
...
+    creddir="${HOME}/.lsst"
+    mkdir -p "${creddir}"
+    if [ -n "${AWS_SHARED_CREDENTIALS_FILE}" ]; then
+        awsname="$(basename ${AWS_SHARED_CREDENTIALS_FILE})"
+        newcreds="${creddir}/${awsname}"
+        chmod 0600 "${newcreds}"
+        cp "${AWS_SHARED_CREDENTIALS_FILE}" "${newcreds}"
+        ORIG_AWS_SHARED_CREDENTIALS_FILE="${AWS_SHARED_CREDENTIALS_FILE}"
+        AWS_SHARED_CREDENTIALS_FILE="${newcreds}"
+        export ORIG_AWS_SHARED_CREDENTIALS_FILE AWS_SHARED_CREDENTIALS_FILE
```
Comment (SimonKrughoff) [inline jupyterlab/runlab.sh:31]: What is the export doing here?  Don't the files just need to exist in the right place? (same below)

**Review by SimonKrughoff — APPROVED** [review]
One probably naive question

File: jupyterlab/runlab.sh (hunk 2 above)
Comment (athornton) [reply jupyterlab/runlab.sh:31]: We want to set the environment variables that are referenced appropriately.  It's bash so I could have made the export and assignment the same statement, but old habits.

Comment (SimonKrughoff) [reply jupyterlab/runlab.sh:31]: OK, fair enough.  Looks good to me.

**Pull request merged**